samples/cellular/modem_shell/src/sock/sock.c

```diff
@@ -805,16 +802,15 @@ static void sock_send_random_data_length(struct sock_info *socket_info)
 		bytes = sock_send(socket_info, socket_info->send_buffer,
 				  strlen(socket_info->send_buffer), false, false);
 
-		if (bytes == -ECANCELED) {
-			socket_info->send_poll = false;
-			break;
-		}
-
-		if (bytes < 0) {
+		if (bytes == -EAGAIN) {
 			/* Wait for socket to allow sending again */
 			socket_info->send_poll = true;
 			return;
+		} else if (bytes < 0) {
+			socket_info->send_poll = false;
+			break;
```

Comment:
Here we handle `-ECANCELED` the same way as other errors, which I think is fine. It ensures the send loop is terminated.

For the `-e` option I kept the handling as-is because the error may be `ENETDOWN`, which indicates that the network is not available. We will soon get a new sockopt `SO_KEEPOPEN` which will keep the socket open during a (temporary) loss of coverage, network PDN re-attach, etc.